docs/shallow_water_equation.md

```diff
@@ -0,0 +1,194 @@
+
+
+## Governing equations
+
+The shallow water equations are given by
+
+\begin{equation}
+\label{eqn:swe}
+\frac{\partial\mathbf{U}}{\partial t} + \frac{\partial \mathbf{E}}{\partial x} + \frac{\partial \mathbf{G}}{\partial x} = \mathbf{S}
+\end{equation}
+
+where 
+
+\begin{align}
+  \mathbf{U}
+  =
+  \begin{bmatrix}
+  h \\[.5em]
+  hu \\[.5em]
+  hv
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{E}
+  =
+  \begin{bmatrix}
+  hu \\[.5em]
+  hu^2 + \frac{1}{2}gh^2 \\[.5em]
+  huv
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{G}
+  =
+  \begin{bmatrix}
+  hv \\[.5em]
+  huv \\[.5em]
+  hv^2 + \frac{1}{2}gh^2
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{S}
+  =
+  \begin{bmatrix}
+  Q_i \\[.5em]
+  -gh\frac{\partial z}{\partial x} - C_D u \sqrt{u^2 + v^2}\\[.5em]
+  -gh\frac{\partial z}{\partial y} - C_D v \sqrt{u^2 + v^2}
+  \end{bmatrix}
+\end{align}
+
+$h$ is the flow depth,
+$u$ is the vertically-averaged velocity in x-direction,
+$v$ is the vertically-averaged velocity in x-direction,
+$z$ is the bed elevation,
+$C_D = g n^2/h^\frac{1}{3}$ is the drag coefficient, and
+$n$ is the Manning's coefficient.
+
+## Spatial discretization
+
+\begin{eqnarray}
+\frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\int_\Omega \frac{\partial\mathbf{E}}{\partial x}  d\Omega + 
+\int_\Omega \frac{\partial\mathbf{G}}{\partial x}  d\Omega +  &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+\frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\oint_{d\Omega} \left( \mathbf{E}dy  - \mathbf{G} dx \right) ds &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+ \frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\int_{d\Omega} \left( \mathbf{F} . \mathbf{n} \right) ds +  &=&
+\int_\Omega \mathbf{S} d\Omega
+\end{eqnarray}
+
+where
+
+$\mathbf{F}$ is the flux vector and
+$\mathbf{n}$ is the outward pointing unit vector to the boundary $\partial\Omega$. 
+The flux normal flux across the face, $\mathbf{F.n}$, is given by
+
+\begin{align}
+  \mathbf{F.n}
+  =
+  \begin{bmatrix}
+  hu_\perp  \\[.5em]
+  huu_\perp + \frac{1}{2}gh^2 \cos\phi \\[.5em]
+  hvu_\perp + \frac{1}{2}gh^2 \sin\phi 
+  \end{bmatrix}
+\end{align}
+
+where $u_\perp$ is the velocity perpendicular to the face given by $u \cos\phi + v \sin\phi$ and
+$\phi$ is the angle between the face normal and the x axis.
+
+
+### Interface flux
+The interface flux, $\mathbf{F}_\perp$, at the face shared by cell $i$ and $j$ are evaluated using Roe's method as 
+
+\begin{equation}
+\mathbf{F.n} \approx \mathbf{F}_\perp^{i,j} =
+\frac{1}{2} \left( \mathbf{F}_\perp^{i} + \mathbf{F}_\perp^{j} - \mathbf{\hat{R}} |\mathbf{\hat{\Lambda}| \mathbf{\Delta}\hat{V}} \right)
+\end{equation}
+
+where
+
+\begin{align}
+  \mathbf{R}
+  =
+  \begin{bmatrix}
+  1 & 0 & 1  \\[.5em]
+  \hat{u} - \hat{c}\cos\phi & -\sin\phi & \hat{u} + \hat{c}\cos\phi \\[.5em]
```

Comment:
I think it should be \hat{a} instead of \hat{c}.

Reply:
fixed